dashboard/src/actions/charts.ts

```diff
@@ -80,6 +95,26 @@ export function fetchCharts(
   };
 }
 
+export function fetchChartCategories(
+  cluster: string,
+  namespace: string,
+  repos: string,
+): ThunkAction<Promise<IChartCategory[]>, IStoreState, null, ChartsAction> {
+  return async dispatch => {
+    dispatch(requestChartsCategories());
+    try {
+      const categories = await Chart.fetchChartCategories(cluster, namespace, repos);
+      if (categories) {
+        dispatch(receiveChartCategories(categories));
+      }
+      return categories;
+    } catch (e) {
+      dispatch(errorChart(new FetchError(e.message)));
```

Comment:
It would be better to create a new type of error for this (e.g. `errorChartCagetories`). Note that if you receive an error here, you need to clean up the state for `categories` in the `reducer`.